src/lib.rs

```diff
@@ -1,27 +1,50 @@
-use std::collections::HashMap;
-
 use borsh::schema::{BorshSchemaContainer, Declaration, Definition, Fields, VariantName};
 use schemars::schema::{RootSchema, Schema};
-use serde::{Deserialize, Serialize};
+use semver::Version;
+use serde::{de, Deserialize, Deserializer, Serialize};
+use std::collections::HashMap;
 
 #[doc(hidden)]
 #[cfg(feature = "__chunked-entries")]
 #[path = "private.rs"]
 pub mod __private;
 
 /// Current version of the ABI schema format.
-pub const SCHEMA_VERSION: &str = env!("CARGO_PKG_VERSION");
+pub const SCHEMA_VERSION_MAJOR: u64 = 0;
+pub const SCHEMA_VERSION_MINOR: u64 = 1;
+pub const SCHEMA_VERSION: &str = "0.1.0";
```

Comment:
How about using `semver::Version` directly?

```suggestion
pub const SCHEMA_VERSION: Version = Version {
    major: 0,
    minor: 1,
    patch: 0,
    pre: semver::Prerelease::EMPTY,
    build: semver::BuildMetadata::EMPTY,
}
```

Reply:
I thought about this, but isn't it weird to expose something with a type from an external crate such as `semver`?

Reply:
Yeah, I suppose we can drop this. We only need major, minor, anyway.